src/include/duckdb/execution/operator/persistent/buffered_csv_reader.hpp

```diff
@@ -84,13 +85,9 @@ struct BufferedCSVReaderOptions {
 	//! Number of sample chunks used for type detection
 	idx_t num_samples = 10;
 	//! The date format to use (if any is specified)
-	StrpTimeFormat date_format;
-	//! Whether or not a date format is specified
-	bool has_date_format = false;
-	//! The timestamp format to use (if any is specified)
-	StrpTimeFormat timestamp_format;
-	//! Whether or not a timestamp format is specified
-	bool has_timestamp_format = false;
+	std::map<LogicalTypeId, StrpTimeFormat> date_format = {{LogicalTypeId::DATE, {}}, {LogicalTypeId::TIMESTAMP, {}}};
```

Comment:
Any specific reason to use map over unordered_map?

Reply:
Yeah, I couldn't see any `hash` definition for `LogicalTypeId` (which is really uint8_t`?). I can poke at that some more.( I first considered using just a `vector` with empty slots, but the `enum` is not dense.`)

Reply:
I think the map is fine, just wondering about the reason :) In general it should only contain two keys anyway.